include/zephyr/llext/loader.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LLEXT_LOADER_H
+#define ZEPHYR_LLEXT_LOADER_H
+
+#include <zephyr/llext/elf.h>
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Loader context for llext
+ * @defgroup llext_loader Loader context for llext
+ * @ingroup llext
+ * @{
+ */
+
+/**
+ * @brief Enum of sections for lookup tables
+ */
+enum llext_section {
+	LLEXT_SECT_TEXT,
+	LLEXT_SECT_DATA,
+	LLEXT_SECT_RODATA,
+	LLEXT_SECT_BSS,
+
+	LLEXT_SECT_REL_TEXT,
+	LLEXT_SECT_REL_DATA,
+	LLEXT_SECT_REL_RODATA,
+	LLEXT_SECT_REL_BSS,
+
+	LLEXT_SECT_SYMTAB,
+	LLEXT_SECT_STRTAB,
+	LLEXT_SECT_SHSTRTAB,
+
+	LLEXT_SECT_COUNT,
+};
+
+/**
+ * @brief Linkable loadable extension loader context
+ */
+struct llext_loader {
+	int (*read)(struct llext_loader *s, void *buf, size_t len);
+	int (*seek)(struct llext_loader *s, size_t pos);
+
+	elf_ehdr_t hdr;
+	elf_shdr_t sects[LLEXT_SECT_COUNT];
+	uint32_t *sect_map;
+	uint32_t sect_cnt;
```

Comment:
missing docstrings

Reply:
These are meant to be private members, I've wrapped them in @cond ignore but please correct me if there's a better way to signal that these are private undocumented members of a public struct